src/effects/effectchainslot.cpp

```diff
@@ -68,6 +68,12 @@ EffectChainSlot::EffectChainSlot(EffectRack* pRack, const QString& group,
 
     connect(&m_channelStatusMapper, SIGNAL(mapped(const QString&)),
             this, SLOT(slotChannelStatusChanged(const QString&)));
+
+    // It is up to skins and controller mappings to do anything useful with this.
+    m_pControlExpanded = new ControlPushButton(ConfigKey(m_group, "expanded"));
```

Comment:
The point of this CO is to have a way for skins and controller mappings to stay in sync with what controls they are exposing. I have made [a generic EffectUnit ControlContainer](https://github.com/mixxxdj/mixxx/pull/1054/files#diff-400922cb0109e689c3d5404810aec4dfR697) for the P32 mapping that makes use of this. It provides a framework for the 3 knobs + 3 buttons + 4th knob/encoder + additional button layout common on most mid-grade and high-grade controllers (such as the Hercules P32, Denon MC6000 & MC4000, Pioneer DDJ-SR & RR, Traktor Kontrol S2/S4/X1, Vestax VCI-400). For controllers with fewer controls like the Pioneer DDJ-SB(2), it will likely make most sense to not use the metaknobs and just use the superknob for the whole EffectUnit with enable buttons for the individual effects within the unit. For the Numark Mixtrack (Pro) 3, (see PR #1014 , ping @djsteph), the metaknobs could be adjusted by holding an effect button down and manipulating the touch strip, with the buttons acting as enable/disable buttons when not held down.